community/kernel/src/main/java/org/neo4j/kernel/impl/factory/PlatformModule.java

```diff
@@ -115,7 +116,12 @@ public PlatformModule( File providedStoreDir, Map<String, String> params, Databa
             @Override
             public DependencyResolver get()
             {
-                return dataSourceManager.getDataSource().getDependencyResolver();
+                NeoStoreDataSource dataSource = dataSourceManager.getDataSource();
+                if ( dataSource == null /* if data source is unavailable */ )
```

Comment:
It's hard to see from a change set why do you need this change, but from a change itself it looks like this is a workaround and it does not look like a safe and correct thing to do.
How/when datasource can be null? Test that cover/show that situation?